.github/workflows/deploy.yml

```diff
@@ -38,12 +40,73 @@ jobs:
     steps:
       - name: Echo Inputs
         run: |
-          echo ref: ${{ github.event.inputs.ref }}
-          echo commit: ${{ github.event.inputs.commit }}
-          echo environment: ${{ github.event.inputs.environment }}
+          echo ref: ${{ inputs.ref }}
+          echo commit: ${{ inputs.commit }}
+          echo environment: ${{ inputs.environment }}
+
+  publish:
+      runs-on: ubuntu-latest
+      if: ${{ github.repository_owner == 'fermyon' }}
+      outputs:
+        oci_ref: ${{ steps.publish-immutable.outputs.oci_ref }}
+      steps:
+        - uses: actions/checkout@v3
+
+        - name: Check out specific ref
+          if: ${{ github.event_name == 'workflow_dispatch' }} && ${{ inputs.ref != ''}}
+          run: git checkout ${{ inputs.ref }}
+
+        - name: Check out specific commit
+          if: ${{ github.event_name == 'workflow_dispatch' }} && ${{ inputs.commit != ''}}
+          run: git checkout ${{ inputs.commit }}
+
+        - name: Setup Node
+          uses: actions/setup-node@v3
+          with:
+            node-version: 16
+
+        - name: Setup Spin
+          uses: fermyon/actions/spin/setup@v1
+          with:
+            github_token: ${{ secrets.GITHUB_TOKEN }}
+            # TODO: update to next release version that includes https://github.com/fermyon/spin/pull/1296
+            version: canary
+
+        - name: Install npm packages
+          run: |
+            npm ci
+
+        # TODO: any other node-y things to do prior to publish? 'npm run styles', etc.?
+
+        - name: Create search index
+          run: |
+            npm run build-index
+
+        - name: Log in to ${{ env.OCI_REGISTRY }}
+          uses: docker/login-action@v2.1.0
+          with:
+            registry: ${{ env.OCI_REGISTRY }}
+            username: fermyon
+            password: ${{ secrets.DOCKERHUB_PAT }}
+
+        - name: Publish Spin app - mutable tag
+          if: ${{ github.ref == 'refs/heads/main' }}
+          run: |
+            spin registry push ${{ env.OCI_REGISTRY }}/${{ env.OCI_IMAGE }}:latest
```

Comment:
there is a Github action for this. (to build/push to OCI). we should give that a shot.

Reply:
Oh right, thank you for the reminder @rajatjindal.  I'll update.